src/seedu/addressbook/commands/SortCommand.java

```diff
@@ -0,0 +1,35 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.data.person.ReadOnlyPerson;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ * List all persons sorted by their name.
+ */
+public class SortCommand extends Command {
+
+    public static final String COMMAND_WORD = "sort";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Displays all persons in the address book as sorted by their name.\n"
+            + "Example: " + COMMAND_WORD;
+
+    @Override
+    public CommandResult execute() {
+        List<ReadOnlyPerson> allPersons = addressBook.getAllPersons().immutableListView();
+        
+        List<ReadOnlyPerson> mutablePersonsList = new ArrayList<>();
+        mutablePersonsList.addAll(allPersons);
+        mutablePersonsList.sort(new Comparator<ReadOnlyPerson>() {
+            @Override
+            public int compare(ReadOnlyPerson o1, ReadOnlyPerson o2) {
+                return o1.getName().fullName.compareTo(o2.getName().fullName);
+            }
```

Comment:
Consider having the ReadOnlyPerson class implementing the Comparable interface instead so that this does not need to be overwritten inside the SortCommand class.